### PR TITLE
Adding an FAQ

### PR DESCRIPTION
**Pull request by arschles**
Adding an FAQ document with answers to a few questions about how KEDA-HTTP is different from other similar projects.

### Checklist

- [x] Commits are signed with Developer Certificate of Origin (DCO)

Fixes https://github.com/kedacore/http-add-on/issues/59
Fixes https://github.com/kedacore/http-add-on/issues/58
Fixes https://github.com/kedacore/http-add-on/issues/57

### CODE REVIEW

**Comment by arschles** [comment]
cc/ @tomkerkhove - PTAL when you have time

File: docs/faq.md [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+- Pure [event-based workloads](https://knative.dev/docs/eventing/). [KEDA core](https://github.com/kedacore/keda), without KEDA-HTTP, can support such workloads natively. If you have a more advanced use case than KEDA core can support, [Dapr](https://dapr.io/) may be a good choice for you.
+- Complex deployment strategies like [blue-green](https://knative.dev/docs/serving/samples/blue-green-deployment/)
+- Supporting other autoscaling mechanisms beyond the built-in [HPA](https://kubernetes.io/docs/tasks/run-application/horizontal-pod-autoscale/)
+
+Additionally, KNative requires a service mesh, while KEDA-HTTP does not out of the box.
+
+## How is this Project Similar or Different from [OpenFaaS](https://www.openfaas.com/)
+
+OpenFaaS and KEDA-HTTP both can deploy and autoscale HTTP workloads onto Kubernetes, but they have several important differences that make them suitable for different use cases:
```
Comment (tomkerkhove) [inline docs/faq.md:28]: As far as I recall, it requires Prometheus as well; can you verify that please?

File: docs/faq.md [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+- The KEDA-HTTP operator can automatically deploy and configure networking and compute resources necessary for an HTTP application to autoscale. Osiris does not have this functionality.
+- Osiris is currently archived in GitHub
+
+## How is this Project Similar or Different from [KNative](https://knative.dev/)?
+
+KNative serving and KEDA-HTTP both have core support for autoscaling, including scale-to-zero of compute workloads. KEDA-HTTP is focused solely on deploying production-grade autoscaling HTTP applications, while KNative builds in additional functionality:
+
+- Pure [event-based workloads](https://knative.dev/docs/eventing/). [KEDA core](https://github.com/kedacore/keda), without KEDA-HTTP, can support such workloads natively. If you have a more advanced use case than KEDA core can support, [Dapr](https://dapr.io/) may be a good choice for you.
+- Complex deployment strategies like [blue-green](https://knative.dev/docs/serving/samples/blue-green-deployment/)
+- Supporting other autoscaling mechanisms beyond the built-in [HPA](https://kubernetes.io/docs/tasks/run-application/horizontal-pod-autoscale/)
```
Comment (tomkerkhove) [inline docs/faq.md:22]: Can you elaborate on this please?

File: docs/faq.md [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
+Additionally, KNative requires a service mesh, while KEDA-HTTP does not out of the box.
```
Comment (tomkerkhove) [inline docs/faq.md:24]: How about this?
```suggestion
Additionally, KNative requires you to use service mesh, while KEDA-HTTP does not (although you can, if you want).
```

**Review by tomkerkhove — COMMENTED** [review]
Just a few thoughts

File: docs/faq.md (hunk 1 above)
Comment (arschles) [reply docs/faq.md:28]: Yup, it requires the [PLONK stack](https://docs.openfaas.com/deployment/#plonk-stack) when running in Kubernetes. I added a note about that in https://github.com/kedacore/http-add-on/pull/71/commits/8182eb580b15e7c55e5401e36d3af0e90077b3d4

**Comment by arschles** [comment]
Thanks @tomkerkhove . I made some edits according to your comments.

File: docs/faq.md [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+
+- Autoscaling concerns are implemented separately from the application resources - `Service`, `Ingress`, `Deployment` and more in KEDA-HTTP. With Osiris, those concerns are baked into each app resource.
+- The KEDA-HTTP operator can automatically deploy and configure networking and compute resources necessary for an HTTP application to autoscale. Osiris does not have this functionality.
+- Osiris is currently archived in GitHub
+
+## How is this Project Similar or Different from [KNative](https://knative.dev/)?
+
+KNative serving and KEDA-HTTP both have core support for autoscaling, including scale-to-zero of compute workloads. KEDA-HTTP is focused solely on deploying production-grade autoscaling HTTP applications, while KNative builds in additional functionality:
+
+- Pure [event-based workloads](https://knative.dev/docs/eventing/). [KEDA core](https://github.com/kedacore/keda), without KEDA-HTTP, can support such workloads natively. If you have a more advanced use case than KEDA core can support, [Dapr](https://dapr.io/) may be a good choice for you.
```
Comment (tomkerkhove) [inline docs/faq.md:20]: Not sure if we should refer to Dapr here as it's not really related?
```suggestion
- Pure [event-based workloads](https://knative.dev/docs/eventing/). [KEDA core](https://github.com/kedacore/keda), without KEDA-HTTP, can support such workloads natively.
```

File: docs/faq.md [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+
+OpenFaaS and KEDA-HTTP both can deploy and autoscale HTTP workloads onto Kubernetes, but they have several important differences that make them suitable for different use cases:
+
+- OpenFaaS requires the use of a CLI to deploy code to production
+- OpenFaaS primarily supports the event-based "functions as a service" pattern. This means:
+  - You deploy code, in a supported language, to handle an HTTP request and OpenFaaS takes care of serving and invoking your code for you
+  - You deploy complete containers with your HTTP server process in them to KEDA-HTTP
+- You don't need to build a container image to deploy code to OpenFaaS, while you do to deploy to KEDA-HTTP
+- OpenFaaS can run either on or off Kubernetes, while KEDA-HTTP is Kubernetes-only
+- OpenFaaS requires several additional components when running in Kubernetes, like Prometheus. The documentation refers to this as the [PLONK stack](https://docs.openfaas.com/deployment/#plonk-stack)
```
Comment (tomkerkhove) [inline docs/faq.md:36]: What about this?
```suggestion
- OpenFaaS requires several additional components when running in Kubernetes, like Prometheus. The documentation refers to this as the [PLONK stack](https://docs.openfaas.com/deployment/#plonk-stack).
  - We do not want to enforce our users to use Prometheus and allow them to choose
```

**Review by tomkerkhove — APPROVED** [review]

File: docs/faq.md (hunk 5 above)
Comment (arschles) [reply docs/faq.md:36]: I'd prefer to keep justification for our various architectural choices out of the FAQ. are you ok with that @tomkerkhove ?

File: docs/faq.md (hunk 4 above)
Comment (arschles) [reply docs/faq.md:20]: Works for me 👍

**Comment by tomkerkhove** [comment]
Removed the build checks as mandatory ones as they don't always run. Based on the trigger, it will block PRs from being merged when they fail.

**Pull request merged**